Calling #vars.getVariable() in thymeleaf-template fails with stacktrace (regression)

With 2.5.1, I can do (for a simple example) something like this:
```
<table>
<tr th:each="var : ${#vars.getVariableNames()}">
            <td th:text="${var}"></td>
            <td th:text="${#vars.getVariable(var)}"></td>
        </tr>
</table>
```
With 2.6.0, this does no longer work; calling #vars.getVariable() fails with the following stacktrace:
```
08:47:49.255 ERROR org.thymeleaf.TemplateEngine - [THYMELEAF][main] Exception processing template "team.html": Exception evaluating OGNL expression: "#vars.getVariable(var)" (template: "team.html" - line 80, col 17)
org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating OGNL expression: "#vars.getVariable(var)" (template: "team.html" - line 80, col 17)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:191)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:95)
        at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:165)
        at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66)
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109)
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138)
        at org.thymeleaf.standard.processor.AbstractStandardExpressionAttributeTagProcessor.doProcess(AbstractStandardExpressionAttributeTagProcessor.java:126)
        at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74)
        at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95)
        at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633)
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314)
        at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205)
        at org.thymeleaf.engine.Model.process(Model.java:282)
        at org.thymeleaf.engine.Model.process(Model.java:290)
        at org.thymeleaf.engine.IteratedGatheringModelProcessable.processIterationModel(IteratedGatheringModelProcessable.java:367)
        at org.thymeleaf.engine.IteratedGatheringModelProcessable.process(IteratedGatheringModelProcessable.java:293)
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleCloseElement(ProcessorTemplateHandler.java:1640)
        at org.thymeleaf.engine.CloseElementTag.beHandled(CloseElementTag.java:139)
        at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136)
        at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:661)
        at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1098)
        at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1067)
        at org.jbake.template.ThymeleafTemplateEngine.renderDocument(ThymeleafTemplateEngine.java:80)
        at org.jbake.template.DelegatingTemplateEngine.renderDocument(DelegatingTemplateEngine.java:61)
        at org.jbake.app.Renderer.render(Renderer.java:207)
        at org.jbake.render.DocumentsRenderer.render(DocumentsRenderer.java:50)
        at org.jbake.app.Oven.bake(Oven.java:155)
        at org.jbake.launcher.Baker.bake(Baker.java:20)
        at org.jbake.launcher.Main.run(Main.java:97)
        at org.jbake.launcher.Main.run(Main.java:82)
        at org.jbake.launcher.Main.main(Main.java:36)
Caused by: ognl.MethodFailedException: Method "getVariable" failed for object 
{all_content=...}[StandardHTMLInliner](team.html)
        at ognl.OgnlRuntime.callAppropriateMethod(OgnlRuntime.java:1551)
        at ognl.ObjectMethodAccessor.callMethod(ObjectMethodAccessor.java:68)
        at ognl.OgnlRuntime.callMethod(OgnlRuntime.java:1615)
        at ognl.ASTMethod.getValueBody(ASTMethod.java:91)
        at ognl.SimpleNode.evaluateGetValueBody(SimpleNode.java:212)
        at ognl.SimpleNode.getValue(SimpleNode.java:258)
        at ognl.ASTChain.getValueBody(ASTChain.java:141)
        at ognl.SimpleNode.evaluateGetValueBody(SimpleNode.java:212)
        at ognl.SimpleNode.getValue(SimpleNode.java:258)
        at ognl.Ognl.getValue(Ognl.java:467)
        at ognl.Ognl.getValue(Ognl.java:431)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.executeExpression(OGNLVariableExpressionEvaluator.java:316)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:170)
        ... 30 common frames omitted
Caused by: java.lang.ClassCastException: org.jbake.app.ContentStore cannot be cast to org.jbake.app.DocumentList
        at org.jbake.template.ThymeleafTemplateEngine$JBakeMap$1$3.loadValue(ThymeleafTemplateEngine.java:117)
        at org.jbake.template.ThymeleafTemplateEngine$JBakeMap$1$3.loadValue(ThymeleafTemplateEngine.java:114)
        at org.thymeleaf.context.LazyContextVariable.getValue(LazyContextVariable.java:84)
        at org.thymeleaf.context.EngineContext.resolveLazy(EngineContext.java:602)
        at org.thymeleaf.context.EngineContext.getVariable(EngineContext.java:194)
        at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at ognl.OgnlRuntime.invokeMethod(OgnlRuntime.java:894)
        at ognl.OgnlRuntime.callAppropriateMethod(OgnlRuntime.java:1539)
        ... 42 common frames omitted
```

--- a/jbake-core/src/main/java/org/jbake/template/AbstractTemplateEngine.java
+++ b/jbake-core/src/main/java/org/jbake/template/AbstractTemplateEngine.java
@@ -1,11 +1,11 @@
 package org.jbake.template;
 
+import org.apache.commons.configuration.Configuration;
+import org.jbake.app.ContentStore;
+
 import java.io.File;
 import java.io.Writer;
 import java.util.Map;
-import org.jbake.app.ContentStore;
-
-import org.apache.commons.configuration.Configuration;
 
 
 /**
@@ -39,5 +39,5 @@ public abstract class AbstractTemplateEngine {
         this.templatesPath = templatesPath;
     }
 
-    public abstract void renderDocument(Map<String,Object> model, String templateName, Writer writer) throws RenderingException;
+    public abstract void renderDocument(Map<String, Object> model, String templateName, Writer writer) throws RenderingException;
 }

--- a/jbake-core/src/main/java/org/jbake/template/ThymeleafTemplateEngine.java
+++ b/jbake-core/src/main/java/org/jbake/template/ThymeleafTemplateEngine.java
@@ -6,7 +6,6 @@ import org.jbake.app.ConfigUtil.Keys;
 import org.jbake.app.ContentStore;
 import org.jbake.app.Crawler;
 import org.jbake.app.Crawler.Attributes;
-import org.jbake.app.DocumentList;
 import org.thymeleaf.TemplateEngine;
 import org.thymeleaf.context.Context;
 import org.thymeleaf.context.LazyContextVariable;
@@ -14,18 +13,22 @@ import org.thymeleaf.templateresolver.FileTemplateResolver;
 
 import java.io.File;
 import java.io.Writer;
-import java.util.*;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.locks.ReentrantLock;
 
 /**
  * <p>A template engine which renders pages using Thymeleaf.</p>
- * <p>
+ *
  * <p>This template engine is not recommended for large sites because the whole model
  * is loaded into memory due to Thymeleaf internal limitations.</p>
- * <p>
+ *
  * <p>The default rendering mode is "HTML", but it is possible to use another mode
  * for each document type, by adding a key in the configuration, for example:</p>
- * <p>
+ *
  * <code>
  * template.feed.thymeleaf.mode=XML
  * </code>
@@ -36,7 +39,6 @@ public class ThymeleafTemplateEngine extends AbstractTemplateEngine {
     private final ReentrantLock lock = new ReentrantLock();
 
     private TemplateEngine templateEngine;
-    private FileTemplateResolver templateResolver;
 
     private String templateMode;
 
@@ -49,7 +51,7 @@ public class ThymeleafTemplateEngine extends AbstractTemplateEngine {
             return;
         }
         templateMode = mode;
-        templateResolver = new FileTemplateResolver();
+        FileTemplateResolver templateResolver = new FileTemplateResolver();
         templateResolver.setPrefix(templatesPath.getAbsolutePath() + File.separatorChar);
         templateResolver.setCharacterEncoding(config.getString(Keys.TEMPLATE_ENCODING));
         templateResolver.setTemplateMode(mode);
@@ -110,11 +112,10 @@ public class ThymeleafTemplateEngine extends AbstractTemplateEngine {
                                     }
                                 };
                             } else {
-                                // All other cases, as far as I know, are document collections
-                                return new LazyContextVariable<DocumentList>() {
+                                return new LazyContextVariable<Object>() {
                                     @Override
-                                    protected DocumentList loadValue() {
-                                        return (DocumentList) extractedValue;
+                                    protected Object loadValue() {
+                                        return extractedValue;
                                     }
                                 };
                             }

--- a/jbake-core/src/test/java/org/jbake/app/template/AbstractTemplateEngineRenderingTest.java
+++ b/jbake-core/src/test/java/org/jbake/app/template/AbstractTemplateEngineRenderingTest.java
@@ -25,17 +25,31 @@ package org.jbake.app.template;
 
 import org.apache.commons.configuration.CompositeConfiguration;
 import org.apache.commons.io.FileUtils;
-import org.jbake.app.*;
+import org.jbake.app.ConfigUtil;
+import org.jbake.app.ContentStore;
+import org.jbake.app.Crawler;
+import org.jbake.app.DBUtil;
+import org.jbake.app.Parser;
+import org.jbake.app.Renderer;
 import org.jbake.model.DocumentTypes;
 import org.jbake.template.ModelExtractors;
 import org.jbake.template.ModelExtractorsDocumentTypeListener;
-import org.junit.*;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
 import java.io.File;
 import java.net.URL;
 import java.nio.charset.Charset;
-import java.util.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
@@ -270,6 +284,27 @@ public abstract class AbstractTemplateEngineRenderingTest {
 
     protected List<String> getOutputStrings(String type) {
         return outputStrings.get(type);
+
+    }
+
+    @Test
+    public void checkDbTemplateModelIsPopulated() throws Exception {
+
+        config.setProperty(ConfigUtil.Keys.PAGINATE_INDEX, true);
+        config.setProperty(ConfigUtil.Keys.POSTS_PER_PAGE, 1);
+
+        outputStrings.put("dbSpan", Arrays.asList("<span>3</span>"));
+
+        db.deleteAllByDocType("post");
+
+        renderer.renderIndexPaging("index.html");
+
+        File outputFile = new File(destinationFolder, "index.html");
+        String output = FileUtils.readFileToString(outputFile, Charset.defaultCharset());
+
+        for (String string : getOutputStrings("dbSpan")) {
+            assertThat(output).contains(string);
+        }
 
     }
 }

--- a/jbake-core/src/test/java/org/jbake/app/template/FreemarkerTemplateEngineRenderingTest.java
+++ b/jbake-core/src/test/java/org/jbake/app/template/FreemarkerTemplateEngineRenderingTest.java
@@ -75,31 +75,10 @@ public class FreemarkerTemplateEngineRenderingTest extends AbstractTemplateEngin
         renderer.renderIndexPaging("index.html");
 
         File paginatedFile = new File(destinationFolder, "index2.html");
-        assertFalse("paginated file is not rendered",paginatedFile.exists());
+        assertFalse("paginated file is not rendered", paginatedFile.exists());
 
         File indexFile = new File(destinationFolder, "index.html");
-        assertTrue("index file exists",indexFile.exists());
-
-    }
-    
-    @Test
-    public void checkDbTemplateModelIsPopulated() throws Exception {
-    	
-        config.setProperty(Keys.PAGINATE_INDEX, true);
-        config.setProperty(Keys.POSTS_PER_PAGE, 1);
-        
-        outputStrings.put("dbSpan", Arrays.asList("<span>3</span>"));
-        
-        db.deleteAllByDocType("post");
-
-        renderer.renderIndexPaging("index.html");
-
-        File outputFile = new File(destinationFolder, "index.html");
-        String output = FileUtils.readFileToString(outputFile, Charset.defaultCharset());
-
-        for (String string : getOutputStrings("dbSpan")) {
-            assertThat(output).contains(string);
-        }
+        assertTrue("index file exists", indexFile.exists());
 
     }
 

--- a/jbake-core/src/test/resources/fixture/groovyMarkupTemplates/index.tpl
+++ b/jbake-core/src/test/resources/fixture/groovyMarkupTemplates/index.tpl
@@ -5,11 +5,12 @@ layout 'layout/main.tpl',
 
                 div(class:"row-fluid marketing"){
                     div(class:"span12"){
-                        published_posts[0..<2].each { post ->
+                        published_posts.each { post ->
                             h4 { a(href:"${post.uri}","${post.title}") }
                             p("${post.date.format("dd MMMM yyyy")} - ${post.body.substring(0, 150)}...")
                         }
                         a(href:"/archive.html",'Archive')
                     }
                 }
+            span("${db.getPublishedPages().size()}")
         }

--- a/jbake-core/src/test/resources/fixture/groovyTemplates/index.gsp
+++ b/jbake-core/src/test/resources/fixture/groovyTemplates/index.gsp
@@ -1,23 +1,26 @@
-<%include "header.gsp"%>
+<% include "header.gsp" %>
 
-      <!--<div class="jumbotron">
-        <h1>Bake your own site!</h1>
-        <p class="lead">Cras justo odio, dapibus ac facilisis in, egestas eget quam. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.</p>
-        <a class="btn btn-large btn-success" href="#">Sign up today</a>
-      </div>
+<!--<div class="jumbotron">
+  <h1>Bake your own site!</h1>
+  <p class="lead">Cras justo odio, dapibus ac facilisis in, egestas eget quam. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.</p>
+  <a class="btn btn-large btn-success" href="#">Sign up today</a>
+</div>
 
-	<hr>-->
+<hr>-->
 
-	<div class="row-fluid marketing">
-		<div class="span12">
-			<%published_posts[0..<2].each { post ->%>
-			<h4><a href="${post.uri}">${post.title}</a></h4>
-			<p>${new java.text.SimpleDateFormat("dd MMMM yyyy", Locale.ENGLISH).format(post.date)} - ${post.body.substring(0, 150)}...</p>
-			<%}%>
-			<a href="/archive.html">Archive</a>
-		</div>
-	</div>
+<div class="row-fluid marketing">
+    <div class="span12">
+        <% published_posts.each { post -> %>
+        <h4><a href="${post.uri}">${post.title}</a></h4>
 
-	<hr>
+        <p>${new java.text.SimpleDateFormat("dd MMMM yyyy", Locale.ENGLISH).format(post.date)} - ${post.body.substring(0, 150)}...</p>
+        <% } %>
+        <a href="/archive.html">Archive</a>
+    </div>
+</div>
 
-<%include "footer.gsp"%>
+<span>${db.getPublishedPages().size()}</span>
+
+<hr>
+
+<% include "footer.gsp" %>

--- a/jbake-core/src/test/resources/fixture/jadeTemplates/index.jade
+++ b/jbake-core/src/test/resources/fixture/jadeTemplates/index.jade
@@ -10,4 +10,6 @@ block content
                     p #{formatter.format(post.date, 'dd MMMM yyyy')} - #{post.body.substring(0, 150)}...
 
             a(href='/archive.html') Archive
+
+    span #{db.getPublishedPages().size()}
     hr

--- a/jbake-core/src/test/resources/fixture/thymeleafTemplates/index.thyme
+++ b/jbake-core/src/test/resources/fixture/thymeleafTemplates/index.thyme
@@ -12,19 +12,22 @@
         <a class="btn btn-large btn-success" href="#">Sign up today</a>
       </div>
 
-	<hr/>-->
+    <hr/>-->
 
-	<div class="row-fluid marketing">
-		<div class="span12">
-			<div th:each="post : ${posts}">
-			<h4><a th:href='${post.uri}' th:text='${post.title}' href="post_uri.html">Post title</a></h4>
-			<p th:text='${#dates.format(post.date,"dd MMMM yyyy") + " - " + #strings.substring(post.body,0, 150)}'>...</p>
-			</div>
-			<a href="/archive.html">Archive</a>
-		</div>
-	</div>
+    <div class="row-fluid marketing">
+        <div class="span12">
+            <div th:each="post : ${posts}">
+            <h4><a th:href='${post.uri}' th:text='${post.title}' href="post_uri.html">Post title</a></h4>
+            <p th:text='${#dates.format(post.date,"dd MMMM yyyy") + " - " + #strings.substring(post.body,0, 150)}'>...</p>
+            </div>
+            <a href="/archive.html">Archive</a>
+        </div>
+    </div>
+
+    <span th:text='${db.getPublishedPages().size()}'>count of published pages</span>
 
 	<hr/>
+
 
     <div th:replace="footer.thyme::footer"></div>
     </body>

